File: spoj py/PTIT124H.py
x=[[0,0],[0,0],[0,0]]
for i in range(0,3):
    s=raw_input()
    f=s.find(' ')
    x[i][0]=int(s[0:f])
    x[i][1]=int(s[f+1::])

for i in range(0,2):
    for j in range(i,3):
        if x[i][0]!=x[j][0] and x[i][1]!=x[j][1]:
            ktx=i
            kty=j

xtd=x[ktx][0]+x[kty][0]
ytd=x[ktx][1]+x[kty][1]
for i in range(0,3):
    if i!=ktx and i!=kty:
        xD=xtd-x[i][0]
        yD=ytd-x[i][1]
print xD,yD

File: PTIT124H.cpp
#include<stdio.h>

int main(void)
{
	int A[3][2],xD,yD,ktx,kty;
	for(int i=0;i<3;i++)
		scanf("%d%d",&A[i][0],&A[i][1]);
	for(int i=0;i<2;i++)
	{
		for(int j=i;j<3;j++)
		{
			if(A[i][0]!=A[j][0] && A[i][1]!=A[j][1])
			{
				ktx=i; kty=j;
			}
		}
	}
	int xtd,ytd;
	xtd=A[ktx][0]+A[kty][0];
	ytd=A[ktx][1]+A[kty][1];
	for(int i=0;i<2;i++)
	{
		if(i!=ktx && i!=kty)
		{
			xD=xtd-A[i][0];
			yD=ytd-A[i][1];
		}
	}
	printf("%d %d",xD,yD);
 return 0;
}

